app/controllers/static_pages_controller.rb

```diff
@@ -1,7 +1,56 @@
-class StaticPagesController < ApplicationController
-  def home
+require 'tsort'
+
+# Topological sorting hash
+class TsortableHash < Hash
+  include TSort
+  alias tsort_each_node each_key
+  def tsort_each_child(node, &block)
+    fetch(node).each(&block)
   end
+end
```

Comment:
If this is going to be used for more than just this controller, you might want to consider putting it in a different file (in `concerns/`)